.github/workflows/build.yml

```diff
@@ -356,7 +356,7 @@ jobs:
                   context: .
 
     webapp-e2e:
-        needs: build-micro-chain
+        needs: [build-cpp, build-micro-chain]
```

Comment:
build-cpp doesn't always run when webapp-e2e needs to run. Probably need to expand what build-micro-chain builds and publish the runners from that.